### ISSUE TITLE
key-spacing: ability to enforce before/afterColon OR align

### ISSUE THREAD

**Issue by mourner** [eslint/eslint#1744]
Currently `key-spacing` rule allows you to enforce either a standard object literal style with zero/one space, or aligned values or colons. The problem is that they're mutually exclusive, so you can't enforce proper 0/1 style while _optionally_ allowing aligned values at the same time:

```js
{
  foo: 'bar',
  helloWorld: 'bla'
}

{
  left:   1,
  right:  2,
  top:    3,
  bottom: 4
}
```

The code above can't be validated using `key-spacing` because it uses both styles, and mixing both styles is pretty common in my experience — some objects look very nice when aligned, while others get less readable, esp. with keys that very in length too much.

P.S. sorry for spamming issues — I'm [switching](https://github.com/mapbox/mapbox-gl-js/pull/957) a relatively large code base from JSHint to ESLint and absolutely love it. :)

**Comment by nzakas** [comment]
I think this is a duplicate of #1728. Can you verify?

**Comment by mourner** [comment]
@nzakas yes, certainly a duplicate! Sorry for not searching properly.
